### PR TITLE
[text analytics] analyze updates

### LINKED ISSUE #16120: [text analytics] analyze endpoint changes
- [ ] rename to `beginAnalyzeBatchTasks`
- [ ] see if makes sense for the task objects to be passed as options bags to regular endpoints
- [ ] make sure usage of `bath`, `job`, `task`, and `operation` is consistent
- [ ] expose job and task metadata on the poller
- [ ] use heterogeneous list of task types as input
- [ ] use heterogeneous list of task type + task error as output
- [ ] add string_index_type as a parameter for all tasks

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes #16120

### CODE REVIEW

File: sdk/textanalytics/azure-ai-textanalytics/README.md [hunk 1]
```diff
@@ -588,26 +619,31 @@ The async versions of the samples (the python sample files appended with `_async
...
-* Batch Analysis: [sample_anayze.py][analyze_sample] ([async version][analyze_sample_async])
+
+- Analyze sentiment: [sample_analyze_sentiment.py][analyze_sentiment_sample] ([async version][analyze_sentiment_sample_async])
+- Recognize entities: [sample_recognize_entities.py][recognize_entities_sample] ([async version][recognize_entities_sample_async])
+- Recognize personally identifiable information: [sample_recognize_pii_entities.py][recognize_pii_entities_sample]([async version][recognize_pii_entities_sample_async])
+- Recognize linked entities: [sample_recognize_linked_entities.py][recognize_linked_entities_sample] ([async version][recognize_linked_entities_sample_async])
+- Extract key phrases: [sample_extract_key_phrases.py][extract_key_phrases_sample] ([async version][extract_key_phrases_sample_async])
+- Detect language: [sample_detect_language.py][detect_language_sample] ([async version][detect_language_sample_async])
+- Healthcare Analysis: [sample_analyze_healthcare.py][analyze_healthcare_sample] ([async version][analyze_healthcare_sample_async])
+- Batch Analysis: [sample_anayze.py][analyze_sample] ([async version][analyze_sample_async])
```
Comment (abhahn) [inline sdk/textanalytics/azure-ai-textanalytics/README.md:634]: Typo in the filename (sample_anayze.py -> sample analyze.py)

File: sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_text_analytics_client.py [hunk 2]
```diff
@@ -707,9 +707,9 @@ def begin_analyze(  # type: ignore
             list[dict[str, str]]
         :param tasks: A list of tasks to include in the analysis.  Each task object encapsulates the parameters
             used for the particular task type.
-        :type tasks: list[Union[~azure.ai.textanalytics.EntitiesRecognitionTask,
-            ~azure.ai.textanalytics.PiiEntitiesRecognitionTask, ~azure.ai.textanalytics.EntityLinkingTask,
-            ~azure.ai.textanalytics.KeyPhraseExtractionTask, ~azure.ai.textanalytics.SentimentAnalysisTask]]
+        :type tasks: list[Union[~azure.ai.textanalytics.EntitiesRecognitionAction,
+            ~azure.ai.textanalytics.PiiEntitiesRecognitionAction, ~azure.ai.textanalytics.EntityLinkingTask,
+            ~azure.ai.textanalytics.KeyPhraseExtractionAction, ~azure.ai.textanalytics.SentimentAnalysisTask]]
```
Comment (abhahn) [inline sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_text_analytics_client.py:712]: I think maybe this must've gotten left over from when we previously had planned to release Sentiment for v3.1-preview.3, so we should probably remove it for now.

File: sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/aio/_text_analytics_client_async.py [hunk 3]
```diff
@@ -709,9 +709,9 @@ async def begin_analyze(  # type: ignore
             list[dict[str, str]]
         :param tasks: A list of tasks to include in the analysis.  Each task object encapsulates the parameters
             used for the particular task type.
-        :type tasks: list[Union[~azure.ai.textanalytics.EntitiesRecognitionTask,
-            ~azure.ai.textanalytics.PiiEntitiesRecognitionTask, ~azure.ai.textanalytics.EntityLinkingTask,
-            ~azure.ai.textanalytics.KeyPhraseExtractionTask, ~azure.ai.textanalytics.SentimentAnalysisTask]]
+        :type tasks: list[Union[~azure.ai.textanalytics.EntitiesRecognitionAction,
+            ~azure.ai.textanalytics.PiiEntitiesRecognitionAction, ~azure.ai.textanalytics.EntityLinkingTask,
+            ~azure.ai.textanalytics.KeyPhraseExtractionAction, ~azure.ai.textanalytics.SentimentAnalysisTask]]
```
Comment (abhahn) [inline sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/aio/_text_analytics_client_async.py:714]: Same as a previous comment for the sync client.

File: sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_text_analytics_client.py [hunk 4]
```diff
@@ -707,9 +707,9 @@ def begin_analyze(  # type: ignore
             list[dict[str, str]]
         :param tasks: A list of tasks to include in the analysis.  Each task object encapsulates the parameters
```
Comment (abhahn) [inline sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_text_analytics_client.py:708]: Should we also rename this to `actions`?

File: sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/aio/_text_analytics_client_async.py [hunk 5]
```diff
@@ -709,9 +709,9 @@ async def begin_analyze(  # type: ignore
             list[dict[str, str]]
         :param tasks: A list of tasks to include in the analysis.  Each task object encapsulates the parameters
```
Comment (abhahn) [inline sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/aio/_text_analytics_client_async.py:710]: Same as another comment for the sync client.

File: sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_text_analytics_client.py (hunk 4 above)
Comment (abhahn) [reply sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_text_analytics_client.py:708]: Also, below in the body of the method I think we will need to group the tasks by type in order to construct the `JobManifest` object.  For some reason github won't let me add a comment directly on it, so check line 750.

File: sdk/textanalytics/azure-ai-textanalytics/samples/async_samples/sample_analyze_async.py [hunk 6]
```diff
@@ -54,12 +54,12 @@ async def analyze_async(self):
...
 
         async with text_analytics_client:
-            poller = await text_analytics_client.begin_analyze(
+            poller = await text_analytics_client.begin_analyze_batch_actions(
                 documents,
                 display_name="Sample Text Analysis",
-                entities_recognition_tasks=[EntitiesRecognitionTask()],
-                pii_entities_recognition_tasks=[PiiEntitiesRecognitionTask()],
-                key_phrase_extraction_tasks=[KeyPhraseExtractionTask()]
+                entities_recognition_tasks=[EntitiesRecognitionAction()],
```
Comment (abhahn) [inline sdk/textanalytics/azure-ai-textanalytics/samples/async_samples/sample_analyze_async.py:60]: I think these were consolidated into a union type with name `tasks` above.

File: sdk/textanalytics/azure-ai-textanalytics/samples/sample_analyze.py [hunk 7]
```diff
@@ -53,12 +53,12 @@ def analyze(self):
...
         ]
 
-        poller = text_analytics_client.begin_analyze(
+        poller = text_analytics_client.begin_analyze_batch_actions(
             documents,
             display_name="Sample Text Analysis",
-            entities_recognition_tasks=[EntitiesRecognitionTask()],
-            pii_entities_recognition_tasks=[PiiEntitiesRecognitionTask()],
-            key_phrase_extraction_tasks=[KeyPhraseExtractionTask()]
+            entities_recognition_tasks=[EntitiesRecognitionAction()],
```
Comment (abhahn) [inline sdk/textanalytics/azure-ai-textanalytics/samples/sample_analyze.py:59]: Same comment as the other sample.

**Comment by iscai-msft** [comment]
closing because there were a bunch of formatting changes that my editor made. easier to start from scratch @abhahn thanks for starting to review!

**Pull request closed without merging**